client/components/site-selector/index.jsx

```diff
@@ -222,38 +225,48 @@ export default React.createClass( {
 	},
 
 	renderRecentSites() {
-		const sites = this.props.sites.getRecentlySelected();
-
-		if ( ! sites || this.state.search || ! this.shouldShowGroups() ) {
-			return null;
+		if ( this.state.search || ! this.shouldShowGroups() ) {
+			return;
 		}
 
-		const recentSites = sites.map( function( site ) {
-			var siteHref;
-
-			if ( this.props.siteBasePath ) {
-				siteHref = this.getSiteBasePath( site ) + '/' + site.slug;
-			}
-
-			const isSelected = this.isSelected( site );
-
-			return (
-				<Site
-					site={ site }
-					href={ siteHref }
-					key={ 'site-' + site.ID }
-					indicator={ this.props.indicator }
-					onSelect={ this.onSiteSelect.bind( this, site.slug ) }
-					isSelected={ isSelected }
-				/>
-			);
-		}, this );
+		const { recentSites } = this.props;
+		if ( ! size( recentSites ) ) {
+			return;
+		}
 
-		if ( ! recentSites ) {
-			return null;
+		const sites = this.props.sites.get();
+		if ( ! sites ) {
+			return;
 		}
 
-		return <div className="site-selector__recent">{ recentSites }</div>;
+		return (
+			<div className="site-selector__recent">
+				{ reduce( sites, ( memo, site ) => {
```

Comment:
>Let's remember to update this later when we have a bit more ported. We should be able to avoid reducing over all sites.

Yeah, I had the same thought while refactoring. If we had sites already indexed by ID (which is the case for Redux where we can use [`getSite`](https://github.com/Automattic/wp-calypso/blob/18a19ca6e918513d0646ef48981f88951adf39d7/client/state/sites/selectors.js#L42-L70)), we'd be able to `map` over the `recentSites` array instead of the other way around. As-is, this way is still much more performant than previously, [where for each recent site we'd filter over the entire sites array](https://github.com/Automattic/wp-calypso/blob/18a19ca6e918513d0646ef48981f88951adf39d7/client/lib/sites-list/list.js#L426-L430).